swift/extractor/visitors/DeclVisitor.h

```diff
@@ -9,247 +9,177 @@
 namespace codeql {
 
 // `swift::Decl` visitor
-// public `visitXXX(swift::XXX* decl)` methods visit concrete declarations
-// private `emitXXX(swift::XXX* decl, TrapLabel<XXXTag> label)` are used to fill the properties
-// related to an abstract `XXX` entity, given the label assigned to the concrete entry
-// In general, `visitXXX/emitXXX` should call `emitYYY` with `YYY` the direct superclass of `XXX`
-// (if not `Decl` itself)
-// TODO: maybe make the above chain of calls automatic with a bit of macro metaprogramming?
 class DeclVisitor : public AstVisitorBase<DeclVisitor> {
  public:
-  using AstVisitorBase<DeclVisitor>::AstVisitorBase;
-
-  void visitFuncDecl(swift::FuncDecl* decl) {
-    auto label = dispatcher_.assignNewLabel(decl);
-    dispatcher_.emit(ConcreteFuncDeclsTrap{label});
-    emitAbstractFunctionDecl(decl, label);
+  void translateFuncDecl(swift::FuncDecl* decl, ConcreteFuncDecl* entry) {
```

Comment:
As a nit, would it make sense to add namespace to the second arguments as well? This way it could be easier to see that we are translating from `swift::Something` to `codeql::Something`.

Reply:
good point!

I was also thinking about the possible name collisions with using the name as is for entries, and thinking about using a more specific prefix like `ConcreteFuncDeclTrap` (which however might be confused with plain single entry traps) or `ConcreteFuncDeclTrapClass`. Makes for longer names, but could save us from possible collisions, and might look clearer in code. Yet another possibility would be to not add any suffix but use two separate namespaces for our application code written by hand (`codeql`) and the generated one (`codeqlgen` or `codeql_gen`).